src/webgpu/api/operation/resource_init/buffer.spec.ts

```diff
@@ -212,3 +215,106 @@ array buffer of getMappedRange() and the GPUBuffer itself have all been initiali
 
     await t.CheckGPUBufferContent(buffer, bufferUsage, expectedData);
   });
+
+g.test('copy_buffer_to_buffer_copy_source')
+  .desc(
+    `Verify when the first usage of a GPUBuffer is being used as the source buffer of
+CopyBufferToBuffer(), the contents of the GPUBuffer have already been initialized to 0.`
+  )
+  .fn(async t => {
+    const bufferSize = 32;
+    const bufferUsage = GPUBufferUsage.COPY_SRC;
+    const buffer = t.device.createBuffer({
+      size: bufferSize,
+      usage: bufferUsage,
+    });
+
+    const expectedData = new Uint8Array(bufferSize);
+    await t.CheckGPUBufferContent(buffer, bufferUsage, expectedData);
+  });
+
+g.test('copy_buffer_to_texture')
+  .desc(
+    `Verify when the first usage of a GPUBuffer is being used as the source buffer of
+CopyBufferToTexture(), the contents of the GPUBuffer have already been initialized to 0.`
+  )
+  .paramsSubcasesOnly(u => u.combine('bufferOffset', [0, 8]))
+  .fn(async t => {
+    const { bufferOffset } = t.params;
+    const textureSize = { width: 8, height: 8, depthOrArrayLayers: 1 };
+    const dstTextureFormat = 'rgba8unorm';
+
+    const dstTexture = t.device.createTexture({
+      size: textureSize,
+      format: dstTextureFormat,
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.COPY_DST,
+    });
+    const layout = getTextureCopyLayout(dstTextureFormat, '2d', [
+      textureSize.width,
+      textureSize.height,
+      textureSize.depthOrArrayLayers,
+    ]);
+    const srcBufferSize = layout.byteLength + bufferOffset;
+
+    const srcBufferUsage = GPUBufferUsage.COPY_SRC;
+    const srcBuffer = t.device.createBuffer({
+      size: srcBufferSize,
+      usage: srcBufferUsage,
+    });
+
+    const stagingBufferUsage = GPUBufferUsage.COPY_DST | GPUBufferUsage.MAP_READ;
+    const stagingBuffer = t.device.createBuffer({
+      size: layout.byteLength,
+      usage: stagingBufferUsage,
+    });
+
+    const encoder = t.device.createCommandEncoder();
+    encoder.copyBufferToTexture(
+      {
+        buffer: srcBuffer,
+        offset: bufferOffset,
+        bytesPerRow: layout.bytesPerRow,
+        rowsPerImage: layout.rowsPerImage,
+      },
+      { texture: dstTexture },
+      textureSize
+    );
+    encoder.copyTextureToBuffer(
+      { texture: dstTexture },
+      { buffer: stagingBuffer, bytesPerRow: layout.bytesPerRow, rowsPerImage: layout.rowsPerImage },
+      textureSize
+    );
+    t.queue.submit([encoder.finish()]);
+
+    // Verify the contents in srcBuffer are all 0.
+    const expectedSrcBufferData = new Uint8Array(srcBufferSize);
+    await t.CheckGPUBufferContent(srcBuffer, srcBufferUsage, expectedSrcBufferData);
+
+    // Verify the texels in dstTexture are all 0.
```

Comment:
I think we can use the `this.expectSingleColor` helper function instead of the staging buffer.